core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1572,6 +1587,8 @@ public function deleteChilds($table, $id, &$delete)
 	 */
 	public function undo()
 	{
+		$this->denyAccessIfDisallowed(self::PERMISSION_MOVE, new DataContainerSubject($this->strTable, $this->intId));
```

Comment:
Not sure no. Undo is very special anyway because proooobably undoing is related to being able to edit a record. But if any relations are gone (e.g. a PID that's gone), voters are probably lost anyway. We could introduce a special `undo` permission if we need it but for now, I've removed it in e02e61e06737a0876795a538ee6cd1712308b752.